### PR TITLE
Draft amnesty week 2024 banners

### PR DESCRIPTION
**Pull request by oetherington**
- Add new ForumEvents collection
- New admin page to manage forum events
- Add hideable frontpage banner during events
- Add smaller posts page banner for relevant posts during an event (not hideable)
- Adds the relevant tag to the frontpage tab bar
- Tweaks to the tag shown on posts items
   - The tag is now clickable
   - The colors match the event

Screenshot 2024-03-11 at 22 27 17

Screenshot 2024-03-11 at 22 27 22

Screenshot 2024-03-11 at 22 24 50

Screenshot 2024-03-11 at 22 24 35

Screenshot 2024-03-11 at 22 29 10

Screenshot 2024-03-11 at 22 30 40

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1206817009201361) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by agnesstenlund** [comment]
Looks good except the image needs to be placed slightly differently–I can send you a smaller version or if there is a way to center it vertically (maybe that's already done?)

**Comment by agnesstenlund** [comment]
Oh and the links need to be underlined.

Otherwise I love this ✨

**Comment by agnesstenlund** [comment]
There should also be a way to dismiss the banner, a small x in the top right corner (not superr important if it takes a while to build)

**Comment by agnesstenlund** [comment]
Also the topics tab should get inverted colors when selected.
Screenshot 2024-03-12 at 14 22 49

 I really love how the topic pills look 👌

**Comment by agnesstenlund** [comment]
Actually–we should invert the topic pill colors in dark mode if that's not too much of a hassle? So it looks like the top one:
Screenshot 2024-03-12 at 14 27 41

**Comment by agnesstenlund** [comment]
(and same with the topic in the topics bar in dark mode, so it looks like this selected)
Screenshot 2024-03-12 at 14 27 23

**Comment by agnesstenlund** [comment]
any chance we could make the fades top earlier? right now much of the picture gets a bit faded, ideally it'd be more isolated on the left hand side. This would be fine to adjust later too though

**Comment by agnesstenlund** [comment]
OK I think I got the fade + placement just perfect. Works for this image, may be weird for other images in the future 🙃

**Comment by Will-Howard** [comment]
@oetherington I'm going to start reviewing the bulk of this in the next hour or so, assuming remaining changes will be relatively small. Does that seem reasonable or is there a lot you want to clean up first (haven't looked yet)?

**Comment by agnesstenlund** [comment]
Thanks @oetherington! Just the frontpage one should be dismissable 🙏

**Comment by oetherington** [comment]
@Will-Howard Thanks - that sounds great!

**Comment by agnesstenlund** [comment]
We also need to link the "Learn more" on the frontpage + "here" on the post page 🙏

**Comment by oetherington** [comment]
@agnesstenlund That doesn't require code changes - it's editable at `/editForumEvent/pSjsmiaWpxLA896uC` (and we'll need to edit it separately on prod once the code is deployed)

**Comment by agnesstenlund** [comment]
OH I see, thank you!

**Comment by oetherington** [comment]
@agnesstenlund Where are you seeing that? I can't find anywhere with a loading state.

**Comment by agnesstenlund** [comment]
when hovering the "Draft Amnesty Week" link in the banner on the Frontpage. I only saw it a couple of times though guess the hover gets saved in the cache after that?

**Review by Will-Howard — APPROVED** [review]
LGTM. The only thing that might be blocking is that malformed frontpage issue that I noticed, maybe try reloading against a cached version of the frontpage with or without the dismiss cookie to see if you can reproduce it. It seems like the sort thing that could just be a one off though so if it can't be reproduced I would assume it won't happen on prod.

**Comment by agnesstenlund** [comment]
not very important though, think we could merge now if the code looks fine @Will-Howard ?

**Comment by agnesstenlund** [comment]
Ok I've a fix for the sm-sized screens. There is a spot where it gets pretty unreadable though so ideally the gradient would also be re-set with a breakpoint. But hopefully this screen-width is uncommon either way:
Screenshot 2024-03-12 at 17 36 21

**Comment by agnesstenlund** [comment]
hmm maybe I actually just made it worse for normal screens now, looks slightly less nice. We could either change the fade based on breakpoints, or just revert my last commit and let the middle sized screens be all green?

File: packages/lesswrong/components/forumEvents/ForumEventFrontpageBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  const hideCookieName = HIDE_FORUM_EVENT_BANNER_PREFIX + currentForumEvent._id;
+  const isHidden = cookies[hideCookieName] === "true";
+  if (isHidden) {
+    return null;
+  }
+
+  const {title, frontpageDescription, bannerImageId, darkColor} = currentForumEvent;
+  const date = formatDate(currentForumEvent);
+
+  const onHide = () => setCookie(hideCookieName, "true");
```
[the remark this answers is not in the document]
Comment (oetherington) [reply packages/lesswrong/components/forumEvents/ForumEventFrontpageBanner.tsx:118]: TIL `useDismissable` exists. Also TIL, I made `useDismissable`.

**Comment by agnesstenlund** [comment]
Alright are we ready to merge? 😃

**Pull request merged**